ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/searchtree/parameters/SearchTreeParameters.java

```diff
@@ -134,6 +135,17 @@ public void setRaLimitationsForSecondPreventive(RaUsageLimits raUsageLimits, Set
         this.raLimitationParameters.put(preventiveInstant, raUsageLimits);
     }
 
+    public void decreaseNumberOfApplicableRemedialActions(Map<Instant, Integer> numberOfAlreadyAppliedRemedialActions) {
+        List<Instant> instants = numberOfAlreadyAppliedRemedialActions.keySet().stream().sorted().toList();
+        instants.forEach(instant -> raLimitationParameters.keySet().forEach(
+            otherInstant -> {
+                if (!otherInstant.comesBefore(instant) && instant.getKind().equals(otherInstant.getKind())) {
+                    raLimitationParameters.get(otherInstant).setMaxRa(raLimitationParameters.get(otherInstant).getMaxRa() - numberOfAlreadyAppliedRemedialActions.get(instant));
```

Comment:
this is not enough, you must decrease other RA limits too : psts, RAs per tso, etc